### ISSUE TITLE
How to add editable cell?

### ISSUE THREAD

**Issue by lexrus** [jung-kurt/gofpdf#91]
I'd like to add some editable fields in my generated PDF with gofpdf. Make them editable in Safari or Chrome as the following screenshot. How can I do that?
<img width="429" alt="screen shot 2016-11-08 at 2 33 27 pm" src="https://cloud.githubusercontent.com/assets/219689/20089235/5e213b92-a5c0-11e6-9ca5-89b94d983aa2.png">

I've already set protection:
``` go
pdf.SetProtection(gofpdf.CnProtectPrint|gofpdf.CnProtectCopy|gofpdf.CnProtectModify, "", "")
```

**Comment by jung-kurt** [comment]
gofpdf does not currently support editable form fields. Fpdf (from which gofpdf is derived) has the following script: [Form filling](http://www.fpdf.org/en/script/script93.php). Is that the kind of functionality you are after?

**Comment by jung-kurt** [comment]
Thanks, @lexrus, for the link. I'll label this issue as an enhancement and with luck find some time to work on it. If you're up to the task, it would make a great pull request!

**Comment by dhalman** [comment]
Hello, and thank you for providing a useful tool for the go community!

Just curious if you have any plans to implement editable fields in gofpdf? It would be awesome to generate forms that can be filled out digitally rather than just when printed :)

thanks again!

**Comment by jung-kurt** [comment]
I agree, this would be a very welcome enhancement. I have not had time to pursue this, but I continue to hope that someone will implement this.

**Comment by dhalman** [comment]
I could attempt a PR with the feature, but I may need some guidance in where to start?

**Comment by jung-kurt** [comment]
Thanks for your interest, @dhalman!

Rather than extending fpdf.go directly, the best approach is to create a new file that contains the extension logic. See layer.go as an example, and notice that it includes methods like layerPutCatalog and layerPutResourceDict to reduce the changes in fpdf.go to some method calls. You may need to add some fields to the Fpdf type; do this is def.go.

Most of the functionality in gofpdf is ported from the fpdf PHP package. It may be that editable field support can come from there as well. The [form filling](http://www.fpdf.org/en/script/script93.php) may be a good start.

Once you get things working, one or more examples in fpdf_test.go provide test coverage and are a real help for people using your extension.

Good luck and don't hesitate to ask questions here along the way.
